src/libutil/suggestions.hh

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "comparator.hh"
+#include "types.hh"
+#include <set>
+
+namespace nix {
+
+int levenshtein_distance(std::string_view first, std::string_view second);
+
+/**
+ * A potential suggestion for the cli interface.
+ */
+class Suggestion {
+public:
+    int distance; // The smaller the better
+    std::string suggestion;
+
+    std::string pretty_print() const;
```

Comment:
```suggestion
    std::string to_string() const;
```
or maybe `operator <<`, then it works in `fmt()` automatically.

Reply:
Is it OK to have `to_string` or `<<` also print some ANSI escape codes?

Reply:
(I’ve made the change anyways since I guess the answer is yes)